### PR TITLE
init ci refactor

### PR DESCRIPTION
**Pull request by KaymeKaydex**
major improvements with CI, allowing you to run integration tests and do static checking of lua

### CODE REVIEW

File: tests/tnt/Makefile [hunk 1]
```diff
@@ -54,7 +63,7 @@ cluster-down:
 # run go tests, minus "-" signs before command allows failures, otherwise cluster-down stage won't run.
 gotest:
 	@echo "${GREEN}STAGE: RUN GOTESTS${NC}"
-	-go test -race -parallel=20 -coverpkg="../../" -coverprofile cover.out -timeout=90s
+	go test -race -parallel=20 -coverpkg="../../" -coverprofile cover.out -timeout=90s
```
Comment (nurzhan-saktaganov) [inline tests/tnt/Makefile:66]: Why we are removing "-" sign?

File: go.mod [hunk 2]
```diff
@@ -35,13 +35,15 @@ require (
 	github.com/subosito/gotenv v1.6.0 // indirect
 	github.com/tarantool/go-iproto v1.0.0 // indirect
 	github.com/vmihailenco/tagparser/v2 v2.0.0 // indirect
+	github.com/wadey/gocovmerge v0.0.0-20160331181800-b5bfa59ec0ad // indirect
```
Comment (nurzhan-saktaganov) [inline go.mod:38]: Are these changes in go.mod, go.sum files required?

File: Makefile [hunk 3]
```diff
@@ -39,4 +45,4 @@ testrace:
 	$(GO_CMD) test -tags='$(BUILD_TAGS)' -race  -timeout=$(EXTENDED_TEST_TIMEOUT) -parallel=20
 
 test/tnt:
-	@$(MAKE) -C ./tests/tnt
+	@$(MAKE) -C ./tests/tnt
```
Comment (nurzhan-saktaganov) [inline Makefile:48]: I guess we need the newline character here ("\n")

File: go.mod (hunk 2 above)
Comment (KaymeKaydex) [reply go.mod:38]: no; i fixed it

File: tests/tnt/Makefile (hunk 1 above)
Comment (KaymeKaydex) [reply tests/tnt/Makefile:66]: cause i hope to get error when  non_existing_command

**Review by nurzhan-saktaganov — APPROVED** [review]

**Pull request merged**